File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
        int res=Integer.MAX_VALUE, mid=0;
        int l=0, r=nums.length;
        while(l<=r){
            mid =(l+r)/2;
            res= Math.min(res,nums[mid]);
            if (nums[mid] > nums[nums.length-1]){
                l=mid+1;
            }
            else r=mid-1;
        }
        return res;
    }
}

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l,r=0,len(nums)-1
        while l<r:
            mid=(l+r)//2
            if nums[mid]>=nums[0]:
                if nums[mid]<=nums[len(nums)-1]:
                    r=mid
                else:
                    l=mid+1
            else :
                r=mid
        return nums[l]
            
        